exp-player/addon/randomizers/permute.js

```diff
@@ -0,0 +1,123 @@
+                        "title": "hippos!"
+                    },
+                    {
+                        "text": "Some more about hippos..."
+                    }
+                ],
+                "kind": "exp-lookit-text"
+            },
+            {
+                "blocks": [
+                    {
+                        "emph": false,
+                        "text": "Let's think about dolphins!",
+                        "title": "dolphins!"
+                    }
+                ],
+                "kind": "exp-lookit-text"
+            }
+        ]
+    }
+}
+*
+* ```
+* @class permute
+*/
+
+var randomizer = function(frameId, frameConfig, pastSessions, resolveFrame) {
+
+    // Data provided to randomizer (properties of frameConfig):
+
+    /**
+     * Unique string identifying this set of frames
+     *
+     * @property {String} id
+     */
+
+    /**
+     * List of frames to be created by this randomizer. Each frame is an
+     * object with any necessary frame-specific properties specified. The
+     * 'kind' of frame can be specified either here (per frame) or in
+     * commonFrameProperties. If a property is defined for a given frame both
+     * in this frame list and in commonFrameProperties, the value in the frame
+     * list will take precedence.
+     *
+     * (E.g., you could include 'kind': 'normal-frame' in
+     * commmonFrameProperties, but for a single frame in frameOptions, include
+     * 'kind': 'special-frame'.)
+     *
+     * @property {Object[]} frameOptions
+     */
+
+    /**
+     * Object describing common parameters to use in EVERY frame created
+     * by this randomizer. Parameter names and values are as described in
+     * the documentation for the frameType used.
+     *
+     * @property {Object} commonFrameProperties
+     */
+
+    /*
+     * Randomize array element order in-place.
+     * Using Durstenfeld shuffle algorithm.
+     */
+    var array = frameConfig.frameOptions.slice();
+    for (var i = array.length - 1; i > 0; i--) {
+        var j = Math.floor(Math.random() * (i + 1));
+        var temp = array[i];
+        array[i] = array[j];
+        array[j] = temp;
+    }
+
+    var thisFrame = {};
+    var frames = [];
+    for (var iFrame = 0; iFrame < array.length; iFrame++) {
+        // Assign parameters common to all frames made by this randomizer
+        thisFrame = {};
+        Object.assign(thisFrame, frameConfig.commonFrameProperties);
+
+        // Assign parameters specific to this frame (allow to override
+        // common parameters assigned above)
+        Object.assign(thisFrame, array[iFrame]);
+
+        thisFrame = resolveFrame(frameId, thisFrame)[0];
+        frames.push(...thisFrame);
+    }
+
+    /**
+     * Parameters captured and sent to the server
+     *
+     * @method conditions
+     * @param {Object[]} frameList the list of frames used, in the final shuffled order
+     */
+    return [frames, {'frameList': array}];
```

Comment:
Thanks for the changes; I'll go ahead and merge once CI is passing.

(on the premise that responding to feedback will only get harder as the next week goes on)

Pretty sure the old randomizers have different arguments because of lack of maintenance, along with some aspects of nesting behavior. (there are some skipped unit tests to that effect if someone ever gets curious) I don't think the other platform project uses those randomizers at all.